sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -1627,3 +1627,149 @@ def sdc_pandas_dataframe_groupby_impl(self, by=None, axis=0, level=None, as_inde
         return init_dataframe_groupby(self, column_id, grouped, sort)
 
     return sdc_pandas_dataframe_groupby_impl
+
+
+def sdc_pandas_dataframe_reset_index_codegen(df_type, all_params, columns, result_name, func_line):
+    """
+    Example of generated implementation:
+        def _df_reset_index_impl(self, level=None, drop=False, inplace=False, col_level=0, col_fill=""):
+          length = len(get_dataframe_data(self, 0))
+          res_index = numpy.arange(length)
+          result_0 = get_dataframe_data(self, 0)
+          return pandas.DataFrame({"index": res_index, "A": result_0}, index=numpy.arange(len(self)))
+   """
+    joined = ', '.join(all_params)
+    func_lines = [f'def _df_reset_index_impl({joined}):']
+    df = all_params[0]
+    func_lines += func_line
+    for i, c in enumerate(columns):
+        result_c = f'result_{i}'
+        func_lines += [
+            f'  result_{i} = get_dataframe_data({df}, {i})'
+        ]
+        result_name.append((result_c, c))
+    data = ', '.join(f'"{column_name}": {column}' for column, column_name in result_name)
+    func_lines += [f'  return pandas.DataFrame({{{data}}}, index=numpy.arange(len(self)))']
+    func_text = '\n'.join(func_lines)
+
+    global_vars = {'pandas': pandas,
+                   'numpy': numpy,
+                   'get_dataframe_data': get_dataframe_data}
+
+    return func_text, global_vars
+
+
+def sdc_pandas_dataframe_reset_index_drop_False(df_type, result_name, func_lines):
+    codegen_ind = df_index_codelines(df_type, True)
+    result_ind = f'res_index'
+    for i in codegen_ind:
+        func_lines += [f'{i}']
+    result_name.append((result_ind, 'index'))
+
+    return result_name, func_lines
+
+
+def sdc_pandas_dataframe_reset_index_impl(self, level=None, drop=False, inplace=False, col_level=0, col_fill=''):
+    all_params = ['self', 'level=None', 'drop=False', 'inplace=False', 'col_level=0', 'col_fill=""']
+    df_func_name = f'_df_reset_index_impl'
+
+    result_name = []
+    func_lines = []
+    drop_val = str(drop)[14:-1]
+    if drop_val == 'False':
+        result_name, func_lines = sdc_pandas_dataframe_reset_index_drop_False(self, result_name, func_lines)
```

Comment:
All this is not needed at all - there should be one function that performs codegen and it's generally a bad idea to produce func text with many functions each passing new own lines to subsequent funcs, like here in func_lines argument. Instead the codegen could have drop param and depending on it's value just add or not add some lines in the resulting func code, e.g. in this case you just need "one function to rule them all":

`    func_text, global_vars = sdc_pandas_dataframe_reset_index_codegen(self, all_params, self.columns,                                                                      drop_literal)`
where:
```
def sdc_pandas_dataframe_reset_index_codegen(df_type, all_params, columns, drop):
    """
    Example of generated implementation:
        def _df_reset_index_impl(self, level=None, drop=False, inplace=False, col_level=0, col_fill=""):
          length = len(get_dataframe_data(self, 0))
          res_index = numpy.arange(length)
          result_0 = get_dataframe_data(self, 0)
          return pandas.DataFrame({"index": res_index, "A": result_0}, index=numpy.arange(len(self)))
   """
    joined = ', '.join(all_params)
    df = all_params[0]
    func_lines = [f'def _df_reset_index_impl({joined}):']
    result_columns = []
    if not drop:
        func_lines += [
            f'  old_index = {df}.index'
        ]
        result_columns.append(('old_index', 'index'))
    for i, c in enumerate(columns):
        result_c = f'result_{i}'
        func_lines += [
            f'  result_{i} = get_dataframe_data({df}, {i})'
        ]
        result_columns.append((result_c, c))
    data = ', '.join(f'"{column_name}": {column}' for column, column_name in result_columns)
    func_lines += [f'  return pandas.DataFrame({{{data}}})']
    func_text = '\n'.join(func_lines)

    global_vars = {'pandas': pandas,
                   'numpy': numpy,
                   'get_dataframe_data': get_dataframe_data}

    return func_text, global_vars
```